### DISCUSSION TITLE
Ignoring mail config functions with null properties

### DISCUSSION THREAD

**Discussion by Synchro** [laravel/framework#39024] [Q&A] [answered]
Say I want to *optionally* BCC some user whenever I send an email:

            Mail::to($user)
                ->bcc(config('mail.bcc'))
                ->send(
                    new Message($user)
                );

If that BCC user email address is not defined in config, the `bcc()` call receives null. Inside the mail instance, the bcc property is set to `null`, which it then attempts to send a message to, and that obviously fails.

The clunky solution to this is (given that I'm not aware of a way to make an inline call conditional):

    if (config('mail.bcc')) {
           Mail::to($user)
                ->bcc(config('mail.bcc'))
                ->send(
                    new Message($user)
                );
    } else {
           Mail::to($user)
                ->send(
                    new Message($user)
                );
    }

*every* time I send a message.

Before I think about making a PR for this, would it be reasonable for Laravel to adjust the mail component so that it ignores null addressees so that we can use the former syntax, i.e. so that `bcc(null)` has the same effect as not calling it at all?

**Comment by Synchro** [comment]
I found a workaround for this. When setting the config value from .env, I I could have it  fall back to the empty value that bcc uses: 

        'bcc'                => env('MAIL_BCC', []),

however, this doesn't work because an empty (rather than missing) value doesn't fall through to the default in `env()`. This does work though:

        'bcc'                => env('MAIL_BCC') ?: [],

This feels "wrong" because the config value should not need to know about the internal workings of the bcc mechanism, but it is at least workable.

**Comment by dennisprudlo** [comment]
What about `->bcc(config('mail.bcc') ?? [])`?

**Reply by Synchro** [reply]
Then I would have to write that every time I wanted to send a message, and because it's relatively unusual syntax, it would be easy to forget to do. Tweaking it in config is much safer, but still not elegant. It also doesn't deal with the case of an empty rather than missing env value, in which case the value obtained from `env()` is an empty string, which doesn't match a `??` comparison.

**Reply by dennisprudlo** [reply]
You can always create a macro to set the bcc with the logic you want (if null or empty string do nothing, else add email to bcc)

**Reply by Synchro** [reply]
I know there are many ways *I* could work around it in my own code, I'm just asking if it seems reasonable to have Laravel act this way by default – what are we currently gaining by allowing `bcc()` to attempt to send to blank addresses?

**Reply by dennisprudlo** [reply]
I can't think of a case where an empty string or null is a legitimate value. So you can create a PR and see if it's get merged.

**Comment by Synchro** [comment, marked as answer]
PR was made and [merged](https://github.com/laravel/framework/pull/39035)!
